src/Microsoft.HttpRepl.IntegrationTests/Commands/UICommandTests.cs

```diff
@@ -0,0 +1,125 @@
+    {
+        public UICommandTests()
+            : base(new UICommand(new UriLauncher()))
+        {
+        }
+
+        [Fact]
+        public void CanHandle_WithNoParseResultSections_ReturnsNull()
+        {
+            bool? result = CanHandle(parseResultSections: string.Empty);
+
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public void CanHandle_WithMoreThanOneParseResultSections_ReturnsNull()
+        {
+            bool? result = CanHandle(parseResultSections: "ui test");
+
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public void CanHandle_WithInvalidName_ReturnsNull()
+        {
+            bool? result = CanHandle(parseResultSections: "test");
+
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public void CanHandle_WithValidName_ReturnsNull()
+        {
+            bool? result = CanHandle(parseResultSections: "ui");
+
+            Assert.True(result);
+        }
+
+        [Fact]
+        public void GetHelpSummary_ReturnsDescription()
+        {
+            string result = GetHelpSummary();
+
+            Assert.Equal(Strings.UICommand_Description, result);
+        }
+
+        [Fact]
+        public void GetHelpDetails_WithInvalidParseResultSection_ReturnsNull()
+        {
+            string result = GetHelpDetails(parseResultSections: "section1");
+
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public void GetHelpDetails_WithMoreThanOneParseResultSections_ReturnsNull()
+        {
+            string result = GetHelpDetails(parseResultSections: "ui section2");
+
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public async Task ExecuteAsync_WithHttpStateBaseAddressSetToNull_WritesToConsoleManagerError()
+        {
+            MockedShellState shellState = new MockedShellState();
+            string parseResultSections = "ui";
+
+            await ExecuteAsyncWithInvalidParseResultSections(parseResultSections, shellState, baseAddress: null);
+
+            VerifyErrorMessageWasWrittenToConsoleManagerError(shellState);
+        }
+
+        [Fact]
+        public async Task ExecuteAsync_WithValidHttpStateBaseAddress_DoesNotThrowException()
+        {
+            MockedShellState shellState = new MockedShellState();
+            ICoreParseResult parseResult = CoreParseResultHelper.Create("ui");
+            HttpClient httpClient = new HttpClient();
+            HttpState httpState = new HttpState(httpClient);
+            Uri uri = new Uri("https://localhost:44366/");
+            httpState.BaseAddress = uri;
+
+            MockUriLauncher mockUriLauncher = new MockUriLauncher(true);
+            UICommand uiCommand = new UICommand(mockUriLauncher);
+
+            var exception = await Record.ExceptionAsync(async () => await uiCommand.ExecuteAsync(shellState, httpState, parseResult, CancellationToken.None));
+
+            Assert.Null(exception);
```

Comment:
If you're not inspecting the exception, why not use Assert.DoesNotThrowAsync() instead of Record.ExceptionAsync()?

Reply:
Hmmm, I don't think the latest version of xunit.assert, version=2.4.1+ has Assert.DoesNotThrowAsync. So leaving this as is. Let me know if I am missing something.

Reply:
Interesting thread on the subject (with alternatives, which includes the Record.Exception option): https://github.com/xunit/xunit/issues/188

Reply:
What do you guys think about doing something like this?

xunit/xunit#188 (comment)

Reply:
Not sure which comment you're referring to (since the link only goes to the issue), but from reading that thread, I'm fine with the Record.Exception pattern.  I agree that Assert.Throws/Assert.DoesNotThrow breaks AAA, and I can appreciate keeping AAA more consistent to improve overall test readability.

Brad does raise a good point though about DoesNotThrow / `Assert.IsNull(exception)`.  This is sort of implicit in every test - that the Act doesn't throw (or else the test would fail anyway).  So it raises a different question - what this test trying to validate?  If you're just validating that it doesn't throw, then a passing implementation could be
```
ExecuteAsync( /* ... */ )
{
    if(IsValid(httpState))
    {
        // do nothing - yay it passes the test
    }
}
// or alternatively
ExecuteAsync( /* ... */ )
{
    if(IsValid(httpState))
    {
        try {
            uriLauncher.Launch( /* ... */ );
        } catch {}
    }
}
// or even further still, this would pass all tests in the suite:
ExecuteAsync( /* ... */ )
{
    if(IsValid(httpState))
    {
        try {
            // do dangerous stuff and mess up the state
        } catch {}
        uriLauncher.Launch( /* ... */ );
    }
}

```

This test as-implemented does not distinguish between these.  Is there something else measurable that can be validated here?

Reply:
Updated the tests here:
https://github.com/aspnet/HttpRepl/commit/1de1b5e283e3554d2e81838bc79195767f3eed4e

The tests verify:
- LaunUriAsync(..) was called
- Exception was recorded if LaunchUriAsync(..) threw exception
- Error was written to console if base address was null